tools/utils/Utils/AppxPackaging/AppxBundleMetadata.cs

```diff
@@ -0,0 +1,186 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AppxBundleMetadata.cs" company="Microsoft">
+//     Copyright (c) Microsoft. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Packaging.Utils.AppxPackaging
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Runtime.InteropServices.ComTypes;
+    using Microsoft.Packaging.Utils.AppxPackagingInterop;
+
+    /// <summary>
+    /// Class that represents appx bundle information.
+    /// </summary>
+    public class AppxBundleMetadata : PackageMetadata
+    {
+        private IAppxBundleManifestReader manifestReader;
+
+        /// <summary>
+        /// Initializes a new instance of the AppxBundleMetadata class.
+        /// </summary>
+        /// <param name="filePath">the path to the appxbundle file</param>
+        public AppxBundleMetadata(string filePath)
+        {
+            this.Initialize(filePath);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the AppxBundleMetadata class.
+        /// </summary>
+        /// <param name="parentBundleInfo">the parent bundle from which this bundle is referenced</param>
+        /// <param name="fileRelativePath">the relative path to the appxbundle file, as referenced 
+        /// from a parent appxbundle file</param>
+        public AppxBundleMetadata(AppxBundleMetadata parentBundleInfo, string fileRelativePath)
+        {
+            this.RelativePath = fileRelativePath;
+            string fullPath = Path.Combine(Path.GetDirectoryName(parentBundleInfo.FilePath), fileRelativePath);
+            this.Initialize(fullPath);
+        }
+
+        /// <summary>
+        /// Gets the relative path this bundle, if it was created by following a reference
+        /// from a parent bundle.
+        /// </summary>
+        public string RelativePath { get; private set; }
+
+        /// <summary>
+        /// Gets the reader for this bundle
+        /// </summary>
+        public IAppxBundleReader AppxBundleReader { get; private set; }
+
+        /// <summary>
+        /// Gets the list of all the external appx packages referenced by the bundle, which belong to the same family.
+        /// </summary>

```

Comment:
We should consider find/replacing these with 'msix'

Reply:
It makes sense to me for comments. Do you also want to change class and namespace names?